### ISSUE TITLE
Fail to compile Everest core

### ISSUE THREAD

**Issue by fulup-bzh** [EVerest/EVerest-archived#49]
Fail to to compile Everest core from scratch:

- initial cmake missing dependencies **(everest-cmake + ev-cli)** 
- I hoped EDM could help solving those dependencies, but it fails with a python error. I used the pip version as described at https://github.com/EVerest/everest-dev-environment/blob/main/dependency_manager /README.md 

I'm running OpenSuSE-15.04 with Python->3.6.15), thank you for your support.
thank you for your support.

**EDM fail in error what ever option is used (here after trace).** 
```
fulup@fulup-desktop:~/Workspace/everest-dev-environment/dependency_manager> edm --version
Traceback (most recent call last):
  File "/home/fulup/.local/bin/edm", line 8, in <module>
    sys.exit(main())
  File "/home/fulup/.local/lib/python3.6/site-packages/edm_tool/__init__.py", line 17, in main
    edm.main(get_parser())
  File "/home/fulup/.local/lib/python3.6/site-packages/edm_tool/__init__.py", line 12, in get_parser
    return edm.get_parser(__version__)
  File "/home/fulup/.local/lib/python3.6/site-packages/edm_tool/edm.py", line 1132, in get_parser
    subparsers = parser.add_subparsers(help='available commands', required=False)
  File "/usr/lib64/python3.6/argparse.py", line 1707, in add_subparsers
    action = parsers_class(option_strings=[], **kwargs)
TypeError: __init__() got an unexpected keyword argument 'required'
```

**Comment by hikinggrass** [comment]
Thank you very much for your bug report!
I think this might be related to the Python version on your system, is it possible for you to try out Python 3.7 or newer?

ev-cli does require Python >= 3.7 anyway, EDM requires python >= 3.6, which probably should be changed.

Btw: here is a quick start guide that might help with other questions about the build process: https://everest.github.io/general/quick_start_guide.html

**Comment by fulup-bzh** [comment]
This is the same the same instruction I follow for EDM. It fails on OpenSuSE-15.04  (last LTS). Tomorrow I will try with Fedora.

**Comment by KodRider** [comment]
During "make install" phase I get the error below : 
 I have tried for two different platforms and got the same result :  
    - Ubuntu 20.04 on Intel 64
    - Debian Bullseye 11 on NXP imx6ul

[ 34%] Linking CXX shared library libframework.so
[ 34%] Built target framework
Scanning dependencies of target manager
[ 34%] Building CXX object _deps/everest-framework-build/src/CMakeFiles/manager.dir/manager.cpp.o
make[2]: *** No rule to make target '_deps/everest-framework-build/src/controller/CMakeFiles/controller-ipc.dir/ipc.cpp.o', needed by '_deps/everest-framework-build/src/manager'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:2897: _deps/everest-framework-build/src/CMakeFiles/manager.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

**Comment by a-w50** [comment]
Hi @KodRider ,

this behaviour seems to be due to cmake cache inconsistencies.  Try a `make clean` or `rm -rf build`

**Comment by KodRider** [comment]
Hi @a-w50,

I have tried your both suggestions but nothing changed.

In fact, a month ago I have succesfully built the images, ran the containers and the simulation on the very same Ubuntu machine.

**Comment by a-w50** [comment]
Interesting ...

did you try a plain `make` before the `make install`?

**Comment by KodRider** [comment]
Did now and the result is same :

**From INTEL with Ubuntu**

[  0%] Built target pugixml-static
[  3%] Built target ryml
[  6%] Built target c4core
[  6%] Built target GENHDR
[ 25%] Built target websockets_shared
[ 26%] Built target nlohmann_json_schema_validator
[ 27%] Built target mqttc
[ 28%] Built target log
[ 29%] Built target fmt
[ 30%] Built target date-tz
[ 34%] Built target framework
make[2]: *** No rule to make target '_deps/everest-framework-build/src/controller/CMakeFiles/controller-ipc.dir/ipc.cpp.o', needed by '_deps/everest-framework-build/src/manager'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:2897: _deps/everest-framework-build/src/CMakeFiles/manager.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
iskoc@iskoc-All-Series:~/checkout/everest-workspace/everest-core/build$ 

**From imx6ul with Debian**

[  0%] Built target pugixml-static
[  3%] Built target ryml
[  6%] Built target c4core
[  6%] Built target GENHDR
[ 25%] Built target websockets_shared
[ 26%] Built target nlohmann_json_schema_validator
[ 27%] Built target mqttc
[ 28%] Built target log
[ 29%] Built target fmt
[ 30%] Built target date-tz
[ 34%] Built target framework
make[2]: *** No rule to make target '_deps/everest-framework-build/src/controller/CMakeFiles/controller-ipc.dir/ipc.cpp.o', needed by '_deps/everest-framework-build/src/manager'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:2895: _deps/everest-framework-build/src/CMakeFiles/manager.dir/all] Error 2
make: *** [Makefile:171: all] Error 2
root@imx6ul-var-dart:~/checkout/everest-workspace/everest-core/build#

**Comment by KodRider** [comment]
I tried to "**make**" directly in the folder "**~/checkout/everest-workspace/everest-core/build/_deps/everest-framework-build/src/controller**"and it worked out well, now continue building with "**make install**"

root@imx6ul-var-dart:~/checkout/everest-workspace/everest-core/build/_deps/everest-framework-build/src/controller# make
Scanning dependencies of target controller-ipc
[  0%] Building CXX object _deps/everest-framework-build/src/controller/CMakeFiles/controller-ipc.dir/ipc.cpp.o

**Comment by a-w50** [comment]
Thanks for this information.  I'll try to figure out, if there is anything I can do about that.  Would it be possible for you, to reproduce the misbehaviour and send me the full logs, after `rm -rf ~/checkout/everest-workspace/everest-core/build/` and a `cmake -S . -B build`, `make -C build -j8`.  By the way, did you use the `-j8` or similar flag when building?

**Comment by KodRider** [comment]
cmake -S . -B build, make -C build -j8 did not work.

I have made a clean build and ended up with the same problem.
Full terminal logs captured and attached..

[log.txt](https://github.com/EVerest/EVerest/files/10409618/log.txt)

**Comment by a-w50** [comment]
Thanks, you need to run the `cmake -S . -B build` in the source folder, and it will create the build folder.  Delete the build folder before! Could you please try that again?
Thanks!

**Comment by KodRider** [comment]
Success with "cmake -S . -B" build and "make -C build -j8" from the source directory.
[success.txt](https://github.com/EVerest/EVerest/files/10409970/success.txt)

So should I always use your method instead of the instructions given : 
  https://everest.github.io/general/quick_start_guide.html#download-and-install
or 
  https://github.com/EVerest/everest-core

**Comment by a-w50** [comment]
These commands should do exactly the same, as the ones in the instructions, they're just a bit shorter.
I assume, that there was some cmake build inconsistency and you probably didn't completely deleted the build folder?
Or can you reproduce the error, when removing the build folder and trying again with `mkdir build && cd build && cmake .. && make -j8`.  If so, please send the complete logs.

**Comment by KodRider** [comment]
I have done 2 builds ( clearing the terminal before ) : 

 1: Success : rm -rf build && mkdir build && cd build && cmake .. && **make -j8**
 2: Fail         : rm -rf build && mkdir build && cd build && cmake .. && **make install**

Only the "make" command different.

[Success.txt](https://github.com/EVerest/EVerest/files/10410368/Success.txt)
[Fail.txt](https://github.com/EVerest/EVerest/files/10410369/Fail.txt)

**Comment by a-w50** [comment]
:1st_place_medal: 
thank you so much.  I guess this will be enough information for me to fix it.

**Comment by a-w50** [comment]
This bug should be fixed by now with EVerest/everest-framework#66
